pkgs/development/compilers/gambit/build.nix

```diff
@@ -7,38 +19,48 @@ stdenv.mkDerivation rec {
-  buildInputs = [ git autoconf bootstrap openssl (makeStaticLibraries openssl)];
+  # TODO: if/when we can get all the library packages we depend on to have static versions,
+  # we could use something like (makeStaticLibraries openssl) to enable creation
+  # of statically linked binaries by gsc.
+  buildInputs = [ git autoconf bootstrap openssl ];
+
+  configureFlags = [
+    "--enable-single-host"
+    "--enable-c-opt=-Os"
+    "--enable-gcc-opts"
+    "--enable-shared"
+    "--enable-absolute-shared-libs" # Yes, NixOS will want an absolute path, and fix it.
+    "--enable-poll"
+    "--enable-openssl"
+    "--enable-default-runtime-options=f8,-8,t8" # Default to UTF-8 for source and all I/O
+    # "--enable-debug" # Nope: enables plenty of good stuff, but also the costly console.log
+    # "--enable-multiple-versions" # Nope, NixOS already does version multiplexing
+    # "--enable-guide"
+    # "--enable-track-scheme"
+    # "--enable-high-res-timing"
+    # "--enable-max-processors=4"
+    # "--enable-multiple-vms"
+    # "--enable-dynamic-tls"
+    # "--enable-multiple-threaded-vms"  # when SMP branch is merged in
+    # "--enable-thread-system=posix"    # default when --enable-multiple-vms is on.
+    # "--enable-profile"
+    # "--enable-coverage"
+    # "--enable-inline-jumps"
+    # "--enable-char-size=1" # default is 4
+  ];
 
   configurePhase = ''
-    options=(
-      --prefix=$out
-      --enable-single-host
-      --enable-c-opt=-O2
-      --enable-gcc-opts
-      --enable-shared
-      --enable-absolute-shared-libs # Yes, NixOS will want an absolute path, and fix it.
-      --enable-poll
-      --enable-openssl
-      --enable-default-runtime-options="f8,-8,t8" # Default to UTF-8 for source and all I/O
-      #--enable-debug # Nope: enables plenty of good stuff, but also the costly console.log
+    export CC=${gcc}/bin/gcc CXX=${gcc}/bin/g++ \
+           CPP=${gcc}/bin/cpp CXXCPP=${gcc}/bin/cpp LD=${gcc}/bin/ld \
+           XMKMF=${coreutils}/bin/false
+    unset CFLAGS LDFLAGS LIBS CPPFLAGS CXXFLAGS
+    ./configure --prefix=$out ${builtins.concatStringsSep " " configureFlags}
 
-      #--enable-multiple-versions # Nope, NixOS already does version multiplexing
-      #--enable-guide
-      #--enable-track-scheme
-      #--enable-high-res-timing
-      #--enable-max-processors=4
-      #--enable-multiple-vms
-      #--enable-dynamic-tls
-      #--enable-multiple-vms
-      #--enable-multiple-threaded-vms  ## when SMP branch is merged in
-      #--enable-thread-system=posix    ## default when --enable-multiple-vms is on.
-      #--enable-profile
-      #--enable-coverage
-      #--enable-inline-jumps
-      #--enable-char-size=1" ; default is 4
-    )
-    ./configure ''${options[@]}
+    # OS-specific paths are hardcoded in ./configure
+    substituteInPlace config.status \
+      --replace /usr/local/opt/openssl/lib "${openssl}/lib" \
+      --replace /usr/local/opt/openssl@1.1/lib "${openssl}/lib"
```

Comment:
```suggestion
      --replace /usr/local/opt/openssl/lib "${openssl.out}/lib" \
      --replace /usr/local/opt/openssl@1.1/lib "${openssl.out}/lib"
```

Without this suggestion:
```
jfelice@crunch /tmp
$ gsc -verbose -exe foo.scm
Parsing:
  "expr"
  "expr"

Compiling:

Dumping:
  #<primitive foo#>

Compilation finished.
/nix/store/1kn7fi3hhi33jms3113riyzwyn2yqpqd-gcc-wrapper-9.2.0/bin/gcc  -Os    -Wno-unused -Wno-write-strings -Wdisabled-optimization -fwrapv -fno-strict-aliasing -fno-trapping-math -fno-math-errno -fschedule-insns2 -fmodulo-sched -freschedule-modulo-scheduled-loops -fomit-frame-pointer -fPIC -fno-common -mpc64   -D___SINGLE_HOST  -I"/nix/store/6w4rkrclxcdivy9jhc25zx6l166g2a13-gambit-4.9.3/include" -c -o "foo.o"  foo.c
/nix/store/1kn7fi3hhi33jms3113riyzwyn2yqpqd-gcc-wrapper-9.2.0/bin/gcc  -Os    -Wno-unused -Wno-write-strings -Wdisabled-optimization -fwrapv -fno-strict-aliasing -fno-trapping-math -fno-math-errno -fschedule-insns2 -fmodulo-sched -freschedule-modulo-scheduled-loops -fomit-frame-pointer -fPIC -fno-common -mpc64   -D___SINGLE_HOST  -I"/nix/store/6w4rkrclxcdivy9jhc25zx6l166g2a13-gambit-4.9.3/include" -c -o "foo_.o"  foo_.c
/nix/store/1kn7fi3hhi33jms3113riyzwyn2yqpqd-gcc-wrapper-9.2.0/bin/gcc     -Wno-unused -Wno-write-strings -Wdisabled-optimization -fwrapv -fno-strict-aliasing -fno-trapping-math -fno-math-errno -fschedule-insns2 -fmodulo-sched -freschedule-modulo-scheduled-loops -fomit-frame-pointer -fPIC -fno-common -mpc64   -rdynamic  -D___SINGLE_HOST  -I"/nix/store/6w4rkrclxcdivy9jhc25zx6l166g2a13-gambit-4.9.3/include"  -o "foo"   foo_.o foo.o "/nix/store/6w4rkrclxcdivy9jhc25zx6l166g2a13-gambit-4.9.3/lib/libgambit.so" -lutil -ldl -lm  -L/nix/store/7i3k8k854al6ci160syrq2xld97kz4ci-openssl-1.1.1d-bin/lib -lssl -lcrypto
/nix/store/ffi3j388c39qbbskfqg1zw0x2lmbzhnk-binutils-2.31.1/bin/ld: cannot find -lssl
/nix/store/ffi3j388c39qbbskfqg1zw0x2lmbzhnk-binutils-2.31.1/bin/ld: cannot find -lcrypto
collect2: error: ld returned 1 exit status
*** ERROR IN ##main -- target link failed while linking "/tmp/foo_.o" "/tmp/foo.o"
```

With this suggestion:
```
jfelice@crunch /tmp
$ gsc -verbose -exe foo.scm
Parsing:
  "expr"
  "expr"

Compiling:

Dumping:
  #<primitive foo#>

Compilation finished.
/nix/store/1kn7fi3hhi33jms3113riyzwyn2yqpqd-gcc-wrapper-9.2.0/bin/gcc  -Os    -Wno-unused -Wno-write-strings -Wdisabled-optimization -fwrapv -fno-strict-aliasing -fno-trapping-math -fno-math-errno -fschedule-insns2 -fmodulo-sched -freschedule-modulo-scheduled-loops -fomit-frame-pointer -fPIC -fno-common -mpc64   -D___SINGLE_HOST  -I"/nix/store/9h6nzjy120q0ls0srfb6fr4iw26b6afm-gambit-4.9.3/include" -c -o "foo.o"  foo.c
/nix/store/1kn7fi3hhi33jms3113riyzwyn2yqpqd-gcc-wrapper-9.2.0/bin/gcc  -Os    -Wno-unused -Wno-write-strings -Wdisabled-optimization -fwrapv -fno-strict-aliasing -fno-trapping-math -fno-math-errno -fschedule-insns2 -fmodulo-sched -freschedule-modulo-scheduled-loops -fomit-frame-pointer -fPIC -fno-common -mpc64   -D___SINGLE_HOST  -I"/nix/store/9h6nzjy120q0ls0srfb6fr4iw26b6afm-gambit-4.9.3/include" -c -o "foo_.o"  foo_.c
/nix/store/1kn7fi3hhi33jms3113riyzwyn2yqpqd-gcc-wrapper-9.2.0/bin/gcc     -Wno-unused -Wno-write-strings -Wdisabled-optimization -fwrapv -fno-strict-aliasing -fno-trapping-math -fno-math-errno -fschedule-insns2 -fmodulo-sched -freschedule-modulo-scheduled-loops -fomit-frame-pointer -fPIC -fno-common -mpc64   -rdynamic  -D___SINGLE_HOST  -I"/nix/store/9h6nzjy120q0ls0srfb6fr4iw26b6afm-gambit-4.9.3/include"  -o "foo"   foo_.o foo.o "/nix/store/9h6nzjy120q0ls0srfb6fr4iw26b6afm-gambit-4.9.3/lib/libgambit.so" -lutil -ldl -lm  -L/nix/store/i8l8gcnz3jya1dr4205gbb92akay3f1f-openssl-1.1.1d/lib -lssl -lcrypto

jfelice@crunch /tmp
$ ./foo
hello, world

```

(it tries to find the libs in the bin package)

Reply:
Yup. I saw that and I fixed it. The naming of package outputs can be confusing.

Reply:
(Thanks!)